test/model/message_list_test.dart

```diff
@@ -238,6 +239,42 @@ void main() {
       ..messages.length.equals(30);
   });
 
+  test('fetchOlder nop during backoff', () => awaitFakeAsync((async) async {
+    final olderMessages = List.generate(5, (i) => eg.streamMessage());
+    final initialMessages = List.generate(5, (i) => eg.streamMessage());
+    await prepare(narrow: const CombinedFeedNarrow());
+    await prepareMessages(foundOldest: false, messages: initialMessages);
+
+    connection.prepare(httpStatus: 400, json: {
+      'result': 'error', 'code': 'BAD_REQUEST', 'msg': 'Bad request'});
+    check(async.pendingTimers).isEmpty();
+    await check(model.fetchOlder()).throws<ZulipApiException>();
+    checkNotified(count: 2);
+    check(model).fetchOlderCoolingDown.isTrue();
+
+    // Drop irrelevant requests with without checking,
+    // as we are only interested in verifying if any request was sent.
+    connection.takeRequests();
+
+    await model.fetchOlder();
+    checkNotNotified();
+    check(model).fetchOlderCoolingDown.isTrue();
+    check(model).fetchingOlder.isFalse();
+    check(connection.lastRequest).isNull();
+
+    // The first backoff is expected to be short enough to complete.
+    async.elapse(const Duration(seconds: 1));
```

Comment:
```suggestion
    // Wait long enough that a first backoff is sure to finish.
    async.elapse(const Duration(seconds: 1));
```

"short enough to complete" sounded to me like saying it's possible to complete it. But the point here is that this specific line definitely does complete it.